src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Interop/DotNetInvoker.ts

```diff
@@ -0,0 +1,103 @@
+﻿import { MethodOptions, DotnetMethodArgumentsList } from './MethodOptions';
+import { platform } from '../Environment';
+import { System_String } from '../Platform/Platform';
+import { getRegisteredFunction } from './RegisteredFunction';
+
+export function invokeDotNetMethod<T>(methodOptions: MethodOptions, ...args: any[]): (T | null) {
+    const method = platform.findMethod(
+        "Microsoft.AspNetCore.Blazor.Browser",
+        "Microsoft.AspNetCore.Blazor.Browser.Interop",
+        "JavaScriptInvoke",
+        "InvokeDotnetMethod");
+
+    const packedArgs = packArguments(args);
+
+    const serializedOptions = platform.toDotNetString(JSON.stringify(methodOptions));
+    const serializedArgs = platform.toDotNetString(JSON.stringify(packedArgs));
+    const serializedResult = platform.callMethod(method, null, [serializedOptions, serializedArgs]);
+
+    if (serializedResult !== null && serializedResult !== undefined && (serializedResult as any) !== 0) {
+        const result = JSON.parse(platform.toJavaScriptString(serializedResult as System_String));
+        return result;
+    }
+
+    return null;
+}
+
+function packArguments(args: any[]): DotnetMethodArgumentsList {
```

Comment:
Instead of specifying an explicit return value type on the signature, could we declare the type on `result` and then let the compiler infer the function's return type? This would provide protection against returning the wrong type accidentally when we edit this function in the future.